### PR TITLE
[utils] Add a Unicode data generator util package

### PR DESCRIPTION
**Pull request by Azoy**
This is a very bare bones package... the targets to generate specific data will be included in the respectful PRs (like: https://github.com/apple/swift/pull/38922 and https://github.com/apple/swift/pull/37864). For now this just sets up the base location for all of these.

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Comment by Azoy** [comment]
@swift-ci please smoke test linux

File: utils/gen-unicode-data/Sources/GenUtils/Files.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public func readFile(_ path: String) -> String {
+  let data = FileManager.default.contents(atPath: path)!
+  return String(decoding: data, as: UTF8.self)
```
Comment (Catfish-Man) [inline utils/gen-unicode-data/Sources/GenUtils/Files.swift:17]: `String(contentsOfFile:encoding:)` does this with slightly less fuss and slightly less indirection

File: utils/gen-unicode-data/Sources/GenUtils/Files.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+public func write(_ data: String, to path: String) {
+  FileManager.default.createFile(
```
Comment (Catfish-Man) [inline utils/gen-unicode-data/Sources/GenUtils/Files.swift:21]: Similarly, there's a (bridged) String method to write directly to a file

**Review by Catfish-Man — APPROVED** [review]
Looks good overall, had a couple trivial suggestions

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Pull request merged**